sign/pending_requests.go

```diff
@@ -0,0 +1,185 @@
+package sign
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/geth/account"
+)
+
+type verifyFunc func(string) (*account.SelectedExtKey, error)
+
+// PendingRequests is a capped container that holds pending signing requests.
+type PendingRequests struct {
+	mu       sync.RWMutex // to guard transactions map
+	requests map[string]*Request
+
+	log log.Logger
+}
+
+// NewRequests creates a new requests list
+func NewRequests() *PendingRequests {
+	logger := log.New("package", "status-go/sign.PendingRequests")
+
+	return &PendingRequests{
+		requests: make(map[string]*Request),
+		log:      logger,
+	}
+}
+
+// Reset is to be used in tests only, as it simply creates new transaction map, w/o any cleanup of the previous one
+func (rs *PendingRequests) Reset() {
+	rs.mu.Lock()
+	defer rs.mu.Unlock()
+
+	rs.requests = make(map[string]*Request)
+}
+
+// Add a new signing request.
+func (rs *PendingRequests) Add(ctx context.Context, meta Meta, completeFunc completeFunc) (*Request, error) {
+	rs.mu.Lock()
+	defer rs.mu.Unlock()
+
+	request := newRequest(ctx, meta, completeFunc)
+	rs.requests[request.ID] = request
+	rs.log.Info("signing request is created", "ID", request.ID)
+
+	go NotifyOnEnqueue(request.ID, meta, ctx)
+
+	return request, nil
+}
+
+// Get returns a signing request by it's ID.
+func (rs *PendingRequests) Get(id string) (*Request, error) {
+	rs.mu.RLock()
+	defer rs.mu.RUnlock()
+
+	if request, ok := rs.requests[id]; ok {
+		return request, nil
+	}
+	return nil, ErrSignReqNotFound
+}
+
+// First returns a first signing request (if exists, nil otherwise). Used in tests.
+func (rs *PendingRequests) First() *Request {
```

Comment:
Again here, I would avoid creating methods just for the purposes of the test. This comment may quickly become invalid when someone uses this method in a different place.

Reply:
I'd keep `r` for `Request` and `rs` for lists/dictionaries, that are `Request*s*`

Reply:
That's my convention too.

Reply:
@adambabik I will fix the comment. we have quite a few places where we basically have test-only methods in this class. I wanted to avoid refactoring tests too much... Maybe when we extract notifications into something, it would be much easier.